### ISSUE TITLE
[Epic] Google Play release: build, sign, submit, roll out

### ISSUE THREAD

**Issue by Strycher** [OffbandMesh/meshcore-client#160]
Part of #159 (Publish Offband Meshcore to Google Play).

## Scope
- Build a signed **release App Bundle** from a clean worktree off `dev`:
  `flutter build appbundle --release --dart-define-from-file=dart_defines.json`
- Confirm signing via the existing release keystore (becomes the Play App Signing **upload key**).
- Verify the AAB: package `app.offband.meshcore`, versionCode, and that `targetSdkVersion` meets Play's current minimum for new apps.

## Out of scope
- Listing copy / privacy policy / declarations (separate child tasks).
- Uploading to Console (Ben's step).

## Acceptance
- Signed `.aab` produced and verified locally; target API level confirmed acceptable for new Play submissions.

**Comment by Strycher** [comment]
## Status update — AAB built locally, but NOT a valid submission build; on HOLD (2026-07-11)

A signed release AAB was produced locally:
- Path: `build/app/outputs/bundle/release/app-release.aab`
- 108 MB · versionName `1.1.2-rc.1` · versionCode `46` · release-signed with the upload key · package `app.offband.meshcore` (verified)

**This build is NOT valid as a submission and is on HOLD:**
1. Built **from dev @ `ed4dce9`**, which **excludes in-flight work** Ben intends to roll into the submission — wrong source.
2. Built **without authorization** — a process/tollgate violation (no explicit "go").

**Nothing was uploaded or pushed.** The AAB is local-only (gitignored `build/` output).

**To complete this task properly:** once Ben decides which in-flight work is included in the first Play submission, rebuild the AAB from that correct, approved source and verify (package, versionCode, signing, target API). Task stays open until then.

**Comment by Strycher** [comment]
Session-end note: content-rating Q4 confirmed **block is in-flight** (being built into the RC for the closed test). This reinforces that the submission AAB must be the in-flight build (incl. block), NOT the dev build built earlier. DarkBasin tracks the in-flight list. Task stays ON HOLD pending Ben's submission-scope decision, then rebuild-from-correct-source + verify.

**Comment by Strycher** [comment]
## AAB build + first-submission procedure (DRAFT for review before building)

Ben wants this reviewed against another session and Gemini BEFORE we build or submit. Nothing gets built until he gives the explicit go. This is the plan for task #160 (the first signed release AAB) plus the first Play upload, which is where the permanent decisions live.

### Two items that are permanent or high-risk (review these hardest)

1. **Play App Signing key is set on the FIRST upload and cannot be changed afterward.**
   - Decision (already made): use OUR OWN release key as the Play app signing key, not a Google-generated one, so Play / GitHub / F-Droid builds stay signature-compatible (users move between channels without uninstalling or losing data).
   - How: export the app signing key from the release keystore (`strycher-personal.jks`) using Google's PEPK (Play Encrypt Private Key) tool, and upload it during Play App Signing setup.
   - RISK: this is irreversible. The exact enrollment flow ("use your own key" vs "let Google generate") must be verified carefully at upload time. If we accept a Google-generated key by mistake, GitHub users cannot update from Play without uninstalling, permanently.

2. **Build source / in-flight scope.**
   - The first submission build must include the intended in-flight work (e.g. the block feature) per DarkBasin's tracked list. Do NOT build from bare `dev` if dev is missing that work.
   - Confirm the exact integrated source with Ben and coordinate with DarkBasin (via Agent Mail) before building.
   - This is the specific mistake already made once: an AAB was built off dev without the in-flight work and without authorization; it is on hold (see #160 earlier note).

### Prerequisites (settle before building)
- [ ] Ben confirms the in-flight scope for the first build; integrated source coordinated with DarkBasin.
- [ ] versionName + versionCode confirmed. pubspec currently `1.1.2-rc.1+46`. versionCode must be a positive integer and must INCREASE on every later update pushed during the test.
- [ ] Gitignored config present in the build tree: `dart_defines.json` (GIPHY key) and `android/key.properties` (release keystore path + passwords).
- [ ] Target SDK meets Play's current minimum for new-app submissions (verify current floor).

### Build steps
1. Check out / worktree the correct integrated source (dev + intended in-flight work).
2. `flutter analyze` clean; `dart format` on any edited files (CI has a format gate that fails on unformatted code).
3. Build: `C:\Dev\flutter-meshcore\bin\flutter.bat build appbundle --release --dart-define-from-file=dart_defines.json` (run from the project dir / correct worktree).
4. Output: `build/app/outputs/bundle/release/app-release.aab`, signed with the release/upload key via the `android/app/build.gradle.kts` signingConfig.
5. Verify the AAB before upload: package `app.offband.meshcore`, correct versionCode/versionName, signed, target SDK acceptable (bundletool dump, or rely on Play's upload-time validation).

### First upload + Play App Signing (the permanent step)
1. Play Console: create the Closed testing release.
2. Play App Signing: choose to provide OUR OWN app signing key; export it from `strycher-personal.jks` via the PEPK tool and upload it. Do NOT accept a Google-generated key. (Verify this flow carefully; it is permanent.)
3. Upload the `.aab`. Add release notes.
4. Complete the closed-test release rollout.

### Closed test + iterate
1. Add the 12+ testers (Google emails or a Google Group), share the opt-in link. The 14-day clock starts once 12 testers are opted in.
2. Push updated builds (higher versionCode, block/in-flight work) as they land. Testers auto-update via Play. The clock is unaffected by updates.

### Production (after 14 continuous days with 12 testers)
1. Apply for production access with the final, complete build.

### Review checklist for the reviewers (other session + Gemini)
- Is the "use our own key" Play App Signing approach correct and safest for a multi-channel (GitHub / F-Droid / Play) OSS app? Any risk in the PEPK export or in the exact enrollment flow we should watch for?
- Is building from a DarkBasin-integrated source the right way to include in-flight work, or is there a cleaner integration path (branch, cherry-pick, worktree)?
- versionCode strategy: `+46` for the first build, then increment for updates. Any conflict with existing tags/build numbers (tags currently run to b58)?
- Target SDK and any current Play policy gate for a first submission that we are missing.
- Anything in the build command, dart-defines, or signing config that could produce a bad or rejectable AAB.
- Anything about starting the closed test on an interim build and updating later that could bite us (rating/data-safety declarations vs the interim build, etc.).

**Comment by Strycher** [comment]
**Tester recruitment update (closed test):**

- **12+ testers now on Ben's mailing list.** Recruitment target met.
- One of the 12 is Ben's own **personal Google account** (strycher@gmail.com), separate from the developer/owner account (support@offband.org). It is a distinct account, so it should count as a valid tester, and it lets Ben test on his own device.
- **Heads-up / buffer:** developer-linked accounts can sometimes be discounted from the 12-tester count. Since 1 of the 12 is Ben, recommend lining up 1-2 more genuinely-external testers so 12 valid opt-ins is guaranteed even if the personal account is not counted. Cheap insurance for the 14-day clock.
- Testers are recruited and **waiting on the opt-in link** from Ben. That link is generated only after the closed test is created, which requires the AAB built + uploaded (per the build plan comment above).

Sequence from here: finish AAB review (other session + Gemini) -> build with correct in-flight scope + own signing key -> upload to closed track -> create closed test -> generate + send opt-in link -> testers opt in + install from Play -> 14-day clock starts once 12 are opted in.

**Comment by Strycher** [comment]
Tester count now **13+ and still recruiting.** Buffer secured: comfortably over 12 even if Ben's personal account is discounted as developer-linked. Testers are fully handled and waiting on the opt-in link. The only remaining gate for the closed test is the build side: AAB plan review (other session + Gemini) -> build with correct in-flight scope + own signing key -> upload -> create closed test -> send opt-in link -> 14-day clock starts.

**Comment by Strycher** [comment]
## Coordination update (DarkBasin, 2026-07-12)

- **Held AAB killed** (the +46 build off stale `ed4dce9`, unauthorized) — deleted per Ben.
- **`dev` is now the complete integrated source** for the submission: all in-flight RC work is merged (block #165 [#164/#168/#169/#170/#172-174], trace/topology #150/#186/#196/#189, GPS #123, queue-drain #51, history-bleed #193/#194, width #151/#154/#155/#156, channel-QR #163). Build the AAB from `dev`, not a stale checkout.
- **versionCode = 59** (Ben's decision: above the b45–b58 GitHub test-build tags, clean monotonic line).
- **Play App Signing = our own key via PEPK** (per this issue's plan) is confirmed correct. A separate AAB runbook draft I wrote had it as Google-managed — that was wrong; corrected to match #160.
- DarkBasin ↔ JadeEagle now coordinating in Agent Mail (thread 108). Open question there: who builds the submission AAB.

**Comment by Strycher** [comment]
**Reconciliation with DarkBasin's runbook confirmed (2026-07-12).** Reviewed DarkBasin's `aab-runbook.md` against the #160 plan; they match.

- **In-flight scope resolved: build from `dev @ tip`.** DarkBasin merged the full RC slate (block #165, trace/topology #150/#186/#196/#189, GPS #123, queue-drain #51, history-bleed #193/#194, width + channel-QR). Closes the "which in-flight work" risk. The stale +46/ed4dce9 AAB is deleted.
- **Own-key Play App Signing via PEPK** (not Google-generated): aligned; DarkBasin corrected their runbook to match #160. Irreversible at first upload; key loss = no update path ever (Bitwarden is the only recovery).
- **versionCode 59** (Ben; above b45-b58 tags, monotonic). Cut `v1.1.2-rc.2+59` (versionName `1.1.2-rc.2`).
- **Division:** DarkBasin = RC compilation (done, `dev` is the base); JadeEagle = Play submission + the AAB build (#160).
- Open coordination point: version bump on protected `dev` (rc.1+46 -> rc.2+59 + tag) — DarkBasin's RC cut or chore(#160); TBD.
- Note: most Play Console setup is already done this session (app created, listing, app-access / target-audience / Families / content-rating / data-safety). Remaining Play-side: version bump -> build from dev -> verify -> PEPK enrollment at upload -> upload -> roll out. Rollout gates: offband.org/privacy live + verify current Play target-SDK floor.
- **Nothing builds until Ben's explicit go.**

**Comment by Strycher** [comment]
Tester list is now a **Google Group: `offband-meshcore-app@googlegroups.com`** (created under the Offband account).
- Ben's 13+ testers added to the group directly (Members > Add members > Add directly), so no self-registration needed; can be opened to "anyone can join/ask" for Discord self-serve growth.
- This group email is the **tester source** entered in Play Console under Closed testing > Testers, which happens when the closed test track is created (post AAB upload). Reusing the collected emails via direct-add, so the earlier email-list work carries over.

**Comment by Strycher** [comment]
**Gemini review findings (2026-07-12) + resolution plan:**

1. **Foreground-service demonstration VIDEO (real, mandatory).** Offband uses a `connectedDevice` foreground service (keeps the radio connected). Targeting Android 14+ requires declaring the FGS type in Play Console (App content > Foreground service permissions) AND linking a demo video. Skipping it = rejection. -> Ben records a 30-60s screen recording (open app -> connect radio over BLE -> persistent connection/notification), uploads unlisted to YouTube, links it in the FGS declaration.
2. **Location justification (lighter than feared).** Foreground location only (BLE scan on Android <=11 + map position); **no background location**; `BLUETOOTH_SCAN` flagged `neverForLocation`. So written justification only, no separate location video. -> JadeEagle drafts the justification text.
3. **GIPHY data sharing.** Privacy policy already discloses GIF search terms -> GIPHY (+ links their policy). Safer to ALSO declare it in Data Safety (App activity / in-app search, shared with GIPHY) rather than rely on the user-initiated exemption, since review inspects network traffic. -> JadeEagle updates the Data Safety guidance.

Actions: JadeEagle drafts FGS + location permission justification text + GIPHY Data Safety update and folds them into the runbook; Ben records the FGS video. The FGS video is a hard submission gate; good that the review caught it pre-build.
